gravitee-apim-e2e/api-test/applications/applications.spec.ts

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright (C) 2015 The Gravitee team (http://gravitee.io)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { beforeAll, describe, expect } from '@jest/globals';
+import { ApplicationsApi } from '@management-apis/ApplicationsApi';
+import {
+  forManagement,
+  forManagementAsAdminUser,
+  forManagementAsApiUser,
+  forManagementAsAppUser,
+  forManagementAsSimpleUser,
+} from '@client-conf/*';
+import { created, noContent, unauthorized } from '../../lib/jest-utils';
+import { ApplicationEntity } from '@management-models/ApplicationEntity';
+import { ApplicationsFaker } from '@management-fakers/ApplicationsFaker';
+
+const managementApplicationsApiAsAdminUser = new ApplicationsApi(forManagementAsAdminUser());
+const managementApplicationsApiAsAppUser = new ApplicationsApi(forManagementAsAppUser());
+const managementApplicationsApiAsApiUser = new ApplicationsApi(forManagementAsApiUser());
+const managementApplicationsApiAsSimpleUser = new ApplicationsApi(forManagementAsSimpleUser());
+const managementApplicationsApiAsUnknownUser = new ApplicationsApi(forManagement({ username: 'unknown', password: 'dummyPwd' }));
+
+const orgId = 'DEFAULT';
+const envId = 'DEFAULT';
+
+let application: ApplicationEntity;
+
+describe('Application tests with authorized users', () => {
+  describe.each`
+    user             | applicationResource
+    ${'ADMIN'}       | ${managementApplicationsApiAsAdminUser}
+    ${'api_user'}    | ${managementApplicationsApiAsApiUser}
+    ${'app_user'}    | ${managementApplicationsApiAsAppUser}
+    ${'simple_user'} | ${managementApplicationsApiAsSimpleUser}
+  `('As $user user', ({ applicationResource }) => {
+    test('should create an application', async () => {
+      application = await created(
+        applicationResource.createApplicationRaw({
+          orgId,
+          envId,
+          newApplicationEntity: ApplicationsFaker.newApplication(),
+        }),
+      );
+      expect(application.status).toEqual('ACTIVE');
+    });
+
+    test('should delete an application', async () => {
```

Comment:
Is it a `test` or an `afterEach` 🤔  Imho it's more an afterEach as your not testing the application deletion in this describe.

Reply:
Well, actually, I do want to test both, the create & the deletion within that describe block.